src/tools/miri/tests/pass-dep/libc/libc-epoll.rs

```diff
@@ -272,7 +272,8 @@ fn test_pointer() {
     assert_eq!(res, 0);
 
     // Register fd[1] with EPOLLIN|EPOLLOUT|EPOLLET
-    let data = MaybeUninit::<u64>::uninit().as_ptr();
+    let data = MaybeUninit::<u64>::uninit();
+    let data = data.as_ptr();
```

Comment:
The fun thing is, this test is running in Miri, so it can't be UB. Obviously the temporary is not actually dropped immediately?

Reply:
Ah never mind, we simply don't care about the pointer actually pointing to live data.

I'm not sure what is the point of this test -- we'll resolve this in Miri.

Reply:
The problem is that here it gets passed to a libc function, which is probably a shim of some sorts.

I tried running this code under miri. It failed:

```rust
fn foo(x: *mut u8) {
    unsafe {
        *x = 123;
    }
}

fn main() {
    let x = core::mem::MaybeUninit::uninit().as_mut_ptr();
    foo(x);
}
```

So either the function does not use the pointer internally (thus not invoking UB), or there's some problem with shimming. Or I am wrong somewhere in my reasoning.